src/Microsoft.Kusto.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -284,8 +284,7 @@ internal void RefreshAzureToken(string ownerUri)
             var response = Instance.ServiceHost.SendRequest(SecurityTokenRequest.Type, requestMessage, true).Result;
             existingConnection.UpdateAzureToken(response.Token);
             
-            existingConnection.TryGetConnection(ConnectionType.Query, out var reliableDataSourceConnection);
-            reliableDataSourceConnection.GetUnderlyingConnection().UpdateAzureToken(response.Token);
```

Comment:
Doing this work within the KustoClient rather finding the client through the connection